### ISSUE TITLE
Compilation fail libv8_libplatform PHP 7.1

### ISSUE THREAD

**Issue by vietnguyen09** [phpv8/v8js#399]
I'm following [this tutorial ](https://github.com/phpv8/v8js/blob/php7/README.Linux.md) tried to install V8 library and v8js but always get this error 

    ....
    checking for libv8_libplatform... configure: error: could not find libv8_libplatform library

I've searched days because I know this is a common issue but no luck so my last choice is to post my problem here and hope someone can help. 

**My PHP & php-config version**

	[root@vnf v8js]# php -v
	PHP 7.1.26 (cli) (built: Jan  9 2019 08:02:33) ( NTS )
	Copyright (c) 1997-2018 The PHP Group
	Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
	You have new mail in /var/spool/mail/root

	[root@vnf v8js]# php-config -version
	Usage: /usr/bin/php-config [OPTION]
	Options:
	  --prefix            [/usr]
	  --includes          [-I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib]
	  --ldflags           []
	  --libs              [-lcrypt   -lresolv -lcrypt -ledit -lncurses -lstdc++ -lz -lrt -lm -ldl -lnsl  -lxml2 -lz -lm -ldl -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -lcrypt -lcrypt ]
	  --extension-dir     [/usr/lib64/php/modules]
	  --include-dir       [/usr/include/php]
	  --man-dir           [/usr/share/man]
	  --php-binary        [/usr/bin/php]
	  --php-sapis         [apache2handler embed fpm  cli phpdbg cgi]
	  --configure-options [--build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --cache-file=../config.cache --with-libdir=lib64 --with-config-file-path=/etc --with-config-file-scan-dir=/etc/php.d --disable-debug --with-pic --disable-rpath --without-pear --with-exec-dir=/usr/bin --with-freetype-dir=/usr --with-png-dir=/usr --with-xpm-dir=/usr --enable-gd-native-ttf --without-gdbm --with-jpeg-dir=/usr --with-openssl --with-system-ciphers --with-zlib --with-layout=GNU --with-kerberos --with-libxml-dir=/usr --with-system-tzdata --with-mhash --enable-dtrace --libdir=/usr/lib64/php --enable-pcntl --enable-opcache --enable-opcache-file --enable-phpdbg --with-imap=shared --with-imap-ssl --enable-mbstring=shared --enable-mbregex --with-gd=shared,/usr --with-gmp=shared --enable-calendar=shared --enable-bcmath=shared --with-bz2=shared --enable-ctype=shared --enable-dba=shared --with-db4=/usr --with-gdbm=/usr --with-tcadb=/usr --enable-exif=shared --enable-ftp=shared --with-gettext=shared --with-iconv=shared --enable-sockets=shared --enable-tokenizer=shared --with-xmlrpc=shared --with-ldap=shared --with-ldap-sasl --enable-mysqlnd=shared --with-mysqli=shared,mysqlnd --with-mysql-sock=/var/lib/mysql/mysql.sock --with-oci8=shared,instantclient,/usr/lib64/oracle/18.3/client64/lib,18.3 --with-pdo-oci=shared,instantclient,/usr,18.3 --with-interbase=shared --with-pdo-firebird=shared --enable-dom=shared --with-pgsql=shared --enable-simplexml=shared --enable-xml=shared --enable-wddx=shared --with-snmp=shared,/usr --enable-soap=shared --with-xsl=shared,/usr --enable-xmlreader=shared --enable-xmlwriter=shared --with-curl=shared,/usr --enable-pdo=shared --with-pdo-odbc=shared,unixODBC,/usr --with-pdo-mysql=shared,mysqlnd --with-pdo-pgsql=shared,/usr --with-pdo-sqlite=shared,/usr --with-pdo-dblib=shared,/usr --with-sqlite3=shared,/usr --enable-json=shared --without-readline --with-libedit --with-pspell=shared --enable-phar=shared --with-mcrypt=shared,/usr --with-tidy=shared,/usr --enable-sysvmsg=shared --enable-sysvshm=shared --enable-sysvsem=shared --enable-shmop=shared --enable-posix=shared --with-unixODBC=shared,/usr --enable-fileinfo=shared --enable-intl=shared --with-icu-dir=/usr --with-enchant=shared,/usr --with-recode=shared,/usr build_alias=x86_64-redhat-linux-gnu host_alias=x86_64-redhat-linux-gnu CFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -fno-strict-aliasing -Wno-pointer-sign CXXFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic]
	  --version           [7.1.26]
	  --vernum            [70126]

**My config.log**
[config.log](https://github.com/phpv8/v8js/files/2913675/config.log)

**UPDATE**

When I run `echo "/opt/v8"  | pecl install v8js-1.0.0` I get this error

        .....................
	/var/tmp/v8js/v8js_array_access.cc: In function 'void v8js_array_access_enumerator(const v8::PropertyCallbackInfo<v8::Array>&)':
	/var/tmp/v8js/v8js_array_access.cc:211:42: error: 'class v8::Object' has no member named 'GetHiddenValue'
	  v8::Local<v8::Value> php_object = self->GetHiddenValue(V8JS_SYM(PHPJS_OBJECT_KEY));
	                                          ^
	make: *** [v8js_array_access.lo] Error 1
	ERROR: `make' failed

Any idea to fix this?

Thanks,

**Comment by jrmgx** [comment]
Similar problem here

```
php -v
PHP 7.2.16-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Mar  7 2019 20:23:29) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.16-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

```
php-config -version
Usage: /usr/bin/php-config [OPTION]
Options:
  --prefix            [/usr]
  --includes          [-I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib ]
  --ldflags           [-L/usr/lib/php/20170718  -L/usr/lib/x86_64-linux-gnu]
  --libs              [-lcrypt   -largon2 -lresolv -lcrypt -lz -lpcre -lrt -lm -ldl -lnsl  -lxml2 -lssl -lcrypto -lsodium -lcrypt -largon2 -largon2 -lcrypt ]
  --extension-dir     [/usr/lib/php/20170718]
  --include-dir       [/usr/include/php/20170718]
  --man-dir           [/usr/share/man]
  --php-binary        [/usr/bin/php7.2]
  --php-sapis         [apache2handler cgi cli fpm ]
  --phpapi            [20170718]
  --configure-options [--includedir=/usr/include --mandir=/usr/share/man --infodir=/usr/share/info --disable-silent-rules --libdir=/usr/lib/x86_64-linux-gnu --libexecdir=/usr/lib/x86_64-linux-gnu --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --enable-cli --disable-cgi --disable-phpdbg --with-config-file-path=/etc/php/7.2/cli --with-config-file-scan-dir=/etc/php/7.2/cli/conf.d --build=x86_64-linux-gnu --host=x86_64-linux-gnu --config-cache --cache-file=/build/php7.2-PL0pac/php7.2-7.2.16/config.cache --libdir=${prefix}/lib/php --libexecdir=${prefix}/lib/php --datadir=${prefix}/share/php/7.2 --program-suffix=7.2 --sysconfdir=/etc --localstatedir=/var --mandir=/usr/share/man --disable-all --disable-debug --disable-rpath --disable-static --with-pic --with-layout=GNU --without-pear --enable-filter --with-openssl=yes --with-password-argon2=/usr --with-pcre-regex=/usr --enable-hash --with-mhash=/usr --enable-libxml --enable-session --with-sodium --with-system-tzdata --with-zlib=/usr --with-zlib-dir=/usr --enable-dtrace --enable-pcntl --with-libedit=shared,/usr build_alias=x86_64-linux-gnu host_alias=x86_64-linux-gnu CFLAGS=-g -O2 -fdebug-prefix-map=/build/php7.2-PL0pac/php7.2-7.2.16=. -fstack-protector-strong -Wformat -Werror=format-security -O2 -Wall -pedantic -fsigned-char -fno-strict-aliasing -g]
  --version           [7.2.16-1+ubuntu18.04.1+deb.sury.org+1]
  --vernum            [70216]
```

v8 version is 7.2.502.25, same problem with 7.1 branches

config.log https://pastebin.com/JHA3GVbf

```
/bin/sh /root/v8js/libtool --mode=compile g++ -Wno-narrowing -std=c++11 -I. -I/root/v8js -DPHP_ATOM_INC -I/root/v8js/include -I/root/v8js/main -I/root/v8js -I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib -I/opt/v8/include -I/opt/v8  -DHAVE_CONFIG_H  -g -O2   -c /root/v8js/v8js_array_access.cc -o v8js_array_access.lo 
libtool: compile:  g++ -Wno-narrowing -std=c++11 -I. -I/root/v8js -DPHP_ATOM_INC -I/root/v8js/include -I/root/v8js/main -I/root/v8js -I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib -I/opt/v8/include -I/opt/v8 -DHAVE_CONFIG_H -g -O2 -c /root/v8js/v8js_array_access.cc  -fPIC -DPIC -o .libs/v8js_array_access.o
In file included from /usr/include/dirent.h:245:0,
                 from /usr/include/php/20170718/Zend/zend_virtual_cwd.h:78,
                 from /usr/include/php/20170718/main/php.h:471,
                 from /root/v8js/php_v8js_macros.h:42,
                 from /root/v8js/v8js_array_access.cc:17:
/root/v8js/v8js_class.h: In function ‘v8js_ctx* v8js_ctx_fetch_object(zend_object*)’:
/root/v8js/v8js_class.h:84:54: warning: offsetof within non-standard-layout type ‘v8js_ctx’ is undefined [-Winvalid-offsetof]
  return (struct v8js_ctx *)((char *)obj - XtOffsetOf(struct v8js_ctx, std));
/root/v8js/v8js_class.h:84:43: note: in expansion of macro ‘XtOffsetOf’
  return (struct v8js_ctx *)((char *)obj - XtOffsetOf(struct v8js_ctx, std));
                                           ^~~~~~~~~~
/root/v8js/v8js_array_access.cc: In function ‘void v8js_array_access_named_getter(v8::Local<v8::String>, const v8::PropertyCallbackInfo<v8::Value>&)’:
/root/v8js/v8js_array_access.cc:226:37: error: no matching function for call to ‘v8::String::Utf8Value::Utf8Value(v8::Local<v8::String>&)’
  v8::String::Utf8Value cstr(property);
                                     ^
In file included from /root/v8js/php_v8js_macros.h:52:0,
                 from /root/v8js/v8js_array_access.cc:17:
/opt/v8/include/v8.h:2878:5: note: candidate: v8::String::Utf8Value::Utf8Value(v8::Isolate*, v8::Local<v8::Value>)
     Utf8Value(Isolate* isolate, Local<v8::Value> obj);
     ^~~~~~~~~
/opt/v8/include/v8.h:2878:5: note:   candidate expects 2 arguments, 1 provided
/root/v8js/v8js_array_access.cc:246:35: error: no matching function for call to ‘v8::Value::ToObject()’
   ret_value = prototype->ToObject()->Get(property);
                                   ^
In file included from /root/v8js/php_v8js_macros.h:52:0,
                 from /root/v8js/v8js_array_access.cc:17:
/opt/v8/include/v8.h:2411:44: note: candidate: v8::MaybeLocal<v8::Object> v8::Value::ToObject(v8::Local<v8::Context>) const
   V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                            ^~~~~~~~
/opt/v8/include/v8.h:2411:44: note:   candidate expects 1 argument, 0 provided
In file included from /opt/v8/include/v8-internal.h:13:0,
                 from /opt/v8/include/v8.h:25,
                 from /root/v8js/php_v8js_macros.h:52,
                 from /root/v8js/v8js_array_access.cc:17:
/opt/v8/include/v8.h:2425:35: note: candidate: v8::Local<v8::Object> v8::Value::ToObject(v8::Isolate*) const
                     Local<Object> ToObject(Isolate* isolate) const);
                                   ^
/opt/v8/include/v8config.h:352:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^~~~~~~~~~
/opt/v8/include/v8.h:2425:35: note:   candidate expects 1 argument, 0 provided
                     Local<Object> ToObject(Isolate* isolate) const);
                                   ^
/opt/v8/include/v8config.h:352:48: note: in definition of macro ‘V8_DEPRECATE_SOON’
 #define V8_DEPRECATE_SOON(message, declarator) declarator
                                                ^~~~~~~~~~
Makefile:208: recipe for target 'v8js_array_access.lo' failed
make: *** [v8js_array_access.lo] Error 1
```

```
uname -a
Linux ns3113308 4.15.0-46-generic #49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by stesie** [comment]
I just released v8js 2.1.1 which should support V8 7.x ... especially current stable line 7.5.xx is known to compile.  Let me know if it still doesn't work

**Comment by nilli29** [comment]
I was getting this error on v8js 2.1.0 (v8 7.1.302.12) but when compiling on v8js 2.1.1.  I am now getting 

Makefile:214: recipe for target 'v8js_convert.lo' failed

Here is my [config.log](https://github.com/phpv8/v8js/files/3474579/config.log)

v8 version 7.6.303.27

```
php -v
----------------------------------
PHP 7.2.20-2+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Jul 25 2019 11:42:55) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.20-2+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

```
php-config -version
----------------------------------
Usage: /usr/bin/php-config [OPTION]
Options:
  --prefix            [/usr]
  --includes          [-I/usr/include/php/20170718 -I/usr/include/php/20170718/main -I/usr/include/php/20170718/TSRM -I/usr/include/php/20170718/Zend -I/usr/include/php/20170718/ext -I/usr/include/php/20170718/ext/date/lib ]
  --ldflags           [-L/usr/lib/php/20170718  -L/usr/lib/x86_64-linux-gnu]
  --libs              [-lcrypt   -largon2 -lresolv -lcrypt -lz -lpcre -lrt -lm -ldl -lnsl  -lxml2 -lssl -lcrypto -lsodium -lcrypt -largon2 -largon2 -lcrypt ]
  --extension-dir     [/usr/lib/php/20170718]
  --include-dir       [/usr/include/php/20170718]
  --man-dir           [/usr/share/man]
  --php-binary        [/usr/bin/php7.2]
  --php-sapis         [apache2handler cgi cli fpm ]
  --phpapi            [20170718]
  --configure-options [--includedir=/usr/include --mandir=/usr/share/man --infodir=/usr/share/info --disable-silent-rules --libdir=/usr/lib/x86_64-linux-gnu --libexecdir=/usr/lib/x86_64-linux-gnu --disable-maintainer-mode --disable-dependency-tracking --prefix=/usr --enable-cli --disable-cgi --disable-phpdbg --with-config-file-path=/etc/php/7.2/cli --with-config-file-scan-dir=/etc/php/7.2/cli/conf.d --build=x86_64-linux-gnu --host=x86_64-linux-gnu --config-cache --cache-file=/build/php7.2-DoofaT/php7.2-7.2.20/config.cache --libdir=${prefix}/lib/php --libexecdir=${prefix}/lib/php --datadir=${prefix}/share/php/7.2 --program-suffix=7.2 --sysconfdir=/etc --localstatedir=/var --mandir=/usr/share/man --disable-all --disable-debug --disable-rpath --disable-static --with-pic --with-layout=GNU --without-pear --enable-filter --with-openssl=yes --with-password-argon2=/usr --with-pcre-regex=/usr --enable-hash --with-mhash=/usr --enable-libxml --enable-session --with-sodium --with-system-tzdata --with-zlib=/usr --with-zlib-dir=/usr --enable-dtrace --enable-pcntl --with-libedit=shared,/usr build_alias=x86_64-linux-gnu host_alias=x86_64-linux-gnu CFLAGS=-g -O2 -fdebug-prefix-map=/build/php7.2-DoofaT/php7.2-7.2.20=. -fstack-protector-strong -Wformat -Werror=format-security -O2 -Wall -pedantic -fsigned-char -fno-strict-aliasing -g]
  --version           [7.2.20-2+ubuntu18.04.1+deb.sury.org+1]
  --vernum            [70220]
```
